subprojects/veins_catch/src/SignalRepresentation.cc

```diff
@@ -0,0 +1,983 @@
+
+                                REQUIRE(signal2.getNumAnalogueModelsApplied() == signal3.getNumAnalogueModelsApplied());
+                                REQUIRE(signal2.getBitrate() == signal3.getBitrate());
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+SCENARIO("Signal Value Access", "[toolbox]")
+{
+    DummySimulation ds(new cNullEnvir(0, nullptr, nullptr)); // necessary so simtime_t works
+    GIVEN("A spectrum with frequencies (1,2,3,4,5,6)")
+    {
+        Freqs freqs = {1, 2, 3, 4, 5, 6};
+
+        SpectrumPtr spectrum = Spectrum::getInstance(freqs);
+
+        WHEN("a signal is constructed and values assigned to frequencies")
+        {
+            Signal signal(spectrum);
+            signal[1] = 4;
+            signal[2] = 3;
+            signal[3] = 2;
+            signal[4] = 1;
+
+            // Access via operators
+            THEN("relative interval is as expected")
+            {
+                REQUIRE(signal.getNumAbsoluteValues() == 6);
+
+                REQUIRE(signal.getRelativeOffset() == 1);
+                REQUIRE(signal.getNumRelativeValues() == 4);
+
+                REQUIRE(signal.getRelativeStart() == 1);
+                REQUIRE(signal.getRelativeEnd() == 5); // Note: This is not 4 as it should support iterations (what about <= in for-loop?)
+            }
+            THEN("frequencies can be addressed by index")
+            {
+                REQUIRE(signal[1] == 4);
+                REQUIRE(signal[2] == 3);
+                REQUIRE(signal[3] == 2);
+                REQUIRE(signal[4] == 1);
+            }
+            THEN("frequencies can be addressed by frequency")
+            {
+                REQUIRE(signal(2) == 4);
+                REQUIRE(signal(3) == 3);
+                REQUIRE(signal(4) == 2);
+                REQUIRE(signal(5) == 1);
+            }
+
+            WHEN("data interval defined")
+            {
+                signal.setDataStart(2);
+                signal.setDataEnd(3);
+
+                THEN("iterating over data interval returns correct values")
+                {
+                    for (uint16_t i = signal.getDataStart(); i < signal.getDataEnd(); i++)
+                        REQUIRE(signal[i] == 5 - i);
+                }
+            }
+        }
+    }
+}
+
+SCENARIO("Invalid Signal Index Access", "[toolbox]")
+{
+    DummySimulation ds(new cNullEnvir(0, nullptr, nullptr)); // necessary so simtime_t works
+    GIVEN("A spectrum with frequencies (1,2,3) and values for a signal with more frequencies (4,3,2,1)")
+    {
+        Freqs freqs = {1, 2, 3};
+        std::vector<double> values = {4, 3, 2, 1};
+        SpectrumPtr spectrum = Spectrum::getInstance(freqs);
+        Signal signal(spectrum);
+        WHEN("An invalid frequency index of the signal gets read")
+        {
+            THEN("An error should be raised (when in DEBUG mode)")
```

Comment:
```suggestion
#ifndef NDEBUG
            THEN("An error should be raised (when in DEBUG mode)")
```

Reply:
Good idea! Though I'll move the `#ifdef` to embrace the whole scenario.

Catch2 would also support to ignore tests that are expected to throw with the `-e` option (see [here](https://github.com/catchorg/Catch2/blob/master/docs/command-line.md#eliding-assertions-expected-to-throw)). But that would require different calls for debug and release mode runs. Thus I think `#ifndef DEBUG` is the better choice.